Cabal/src/Distribution/Simple/PackageDescription.hs

```diff
@@ -65,10 +69,35 @@ parseString
     -> IO a
 parseString parser verbosity name bs = do
     let (warnings, result) = runParseResult (parser bs)
-    traverse_ (warn verbosity . showPWarning name) warnings
+    traverse_ (warn verbosity . showPWarning name) (flattenDups verbosity warnings)
     case result of
         Right x -> return x
         Left (_, errors) -> do
             traverse_ (warn verbosity . showPError name) errors
             die' verbosity $ "Failed parsing \"" ++ name ++ "\"."
 
+-- Collapse duplicate experimental feature warnings into single warning, with
```

Comment:
```suggestion
-- | Collapse duplicate experimental feature warnings into single warning, with
```

Because we :heart: haddock. Also, tooling can show you the haddock comments on hover.